back-end/controllers/record-controller.js

```diff
@@ -76,23 +151,58 @@ const recordController = {
   // Delete the record
   // URL: delete /records/:rid
   deleteRecord: async (req, res, next) => {
-    try {
-      const recordId = Number(req.params.rid)
-      const userId = req.user.id
+    const recordId = Number(req.params.rid)
+    const userId = req.user.id
+
+    // Check if there is record id in req.params
+    if (!recordId) {
+      return res.status(400).json({
+        type: 'Delete record failed',
+        title: 'Require reocrd id',
+        field_errors: {
+          recordId: 'required',
+        },
+      })
+    }
 
+    // Check data type
+    if (typeof recordId !== 'number') {
+      return res.status(400).json({
+        type: 'Delete record failed',
+        title: 'Incorrect data type',
+        field_errors: {
+          recordId: 'number',
+        },
+      })
+    }
+    try {
       // Check if the record is in database
       const theRecord = await prisma.record.findUnique({
         where: {
           id: recordId,
         },
       })
-      if (!theRecord) return res.status(400).json('the record does not exist')
+      // !theRecord: {}
+      if (theRecord === 'null') {
```

Comment:
Same as Line: 118

Reply:
On line 118 & 186, I have no idea why if I couldn't find `theRecord` with `recordId`, the outcome value is `null`, rather than `{}`. I am sure it was `{}` yesterday!